test/iterators/dfs.jl

```diff
@@ -0,0 +1,44 @@
+@testset "DFSIterator" begin
```

Comment:
It would be good to test edge cases: empty graph, graph with 2 connected components

Reply:
graph with 2 connected component is already tested

Reply:
but adding an empty graph seems good

Reply:
added it